server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -747,13 +747,14 @@ private void mergeAndPush()
     for (Map.Entry<Long, Sink> entry : sinks.entrySet()) {
       final Long intervalStart = entry.getKey();
       if (intervalStart < minTimestamp) {
-        log.info("Adding entry[%s] for merge and push.", entry);
+        log.info("Adding entry [%s] for merge and push.", entry);
         sinksToPush.add(entry);
       } else {
         log.warn(
-            "[%s] < [%s] Skipping persist and merge.",
+            "[%s] >= [%s] Skipping persist and merge for entry [%s]",
```

Comment:
while we are at it, can we update the logging on line 744 as well?
 log.info("Found [%,d] sinks. minTimestamp [%s]", sinks.size(), minTimestampAsDate);

I think it'd be nice for this to say something along the lines of:
 log.info("Found [%,d] segments. Attempting to hand off segments that start before [%s].", sinks.size(), minTimestampAsDate);